### ISSUE TITLE
The error "MPV Shim Login Failed" frequently occurs when logging into Jellyfin MPV Desktop on Jellyfin 10.6.x

### ISSUE THREAD

**Issue by whalehub** [iwalton3/jellyfin-mpv-shim#99]
**Describe the bug**
While we can successfully log into Jellyfin from my website and virtually any of the available Jellyfin apps (including Jellyfin MPV Shim), my users and I have been experiencing a strange bug in Jellyfin MPV Desktop ever since the release of Jellyfin 10.6.x. It should be noted that clients which logged into Jellyfin MPV Desktop prior to the release of Jellyfin 10.6.x continue to work fine.

In roughly 9/10 login attempts (where the folder `%APPDATA%/jellyfin-mpv-shim` is deleted between each attempt), Jellyfin MPV Desktop will display the error message "MPV Shim Login Failed" a few seconds after logging in.

At first, I thought that this was a visual bug because the login attempt in the embedded web client *does* succeed on a surface level ("Jellyfin MPV Desktop" is successfully added to the "Devices" tab in Jellyfin and I can browse all parts of Jellyfin like I would in the browser). However, when I try playing back media, I receive the error message "Playback Error: This client isn't compatible with the media and the server isn't sending a compatible media format."

I enabled the debug logging feature in the configuration file and noticed that for all login attempts where the "MPV Shim Login Failed" error message is displayed, the Python Jellyfin API client timed out after five seconds and reported that the login failed. There was also no `cred.json` file generated in the AppData folder.

```
2020-08-12 06:22:58,278 [   ERROR] JELLYFIN.jellyfin_apiclient_python.api: HTTPSConnectionPool(host='<redacted>', port=443): Read timed out. (read timeout=5)
2020-08-12 06:22:58,278 [    INFO] JELLYFIN.jellyfin_apiclient_python.connection_manager: Failed to login as `<redacted>`
```

As you can see in the attached debug logs, this error message is absent for the rare successful login attempt. I've verified that in those cases, a `cred.json` file is successfully generated in the AppData folder and that playing back media works fine.

**To Reproduce**
There are honestly far too many potential variables involved for me to provide a set of reproducible instructions, but my hypothesis is that this issue is caused by the thread locking bug in Jellyfin 10.6.x which has supposedly already been fixed upstream but has not yet been released as part of a new Jellyfin version.

**Expected behavior**
I expect the timeout to be significantly longer than five seconds to account for slow connections, perhaps closer to 30 seconds.

@iwalton3 If you can generate a Jellyfin MPV Desktop debug build that was compiled with a [30 second timeout](https://github.com/iwalton3/jellyfin-apiclient-python/blob/master/jellyfin_apiclient_python/api.py#L46), we could see if that's enough to resolve this issue.

**Screenshots**

![error](https://xbb.datahoarder.dev/VUSu3/XAyagaSi80.png/raw)

![playback](https://xbb.datahoarder.dev/VUSu3/QOHEHuLi31.png/raw)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version 1.7.0

**Error Messages**
[failed-login.txt](https://github.com/iwalton3/jellyfin-mpv-shim/files/5060744/failed-login.txt)
[successful-login.txt](https://github.com/iwalton3/jellyfin-mpv-shim/files/5060745/successful-login.txt)

**Comment by iwalton3** [comment]
This is a desktop debug build (so you don't have to export logs to see errors) with the client timeout set to 30 seconds. I hope this helps!
https://iwalton.com/ushare/jellyfin-mpv-shim_1.7.0_timeout30.exe

**Comment by iwalton3** [comment]
To provide more of a detail of what causes this, logging into the desktop client actually creates *two* sessions. This is unfortunately core to the way the desktop app works and cannot easily be changed (at least until the web client is done with ongoing rewrite).

When you log into the desktop client, I [intercept the API call here](https://github.com/iwalton3/jellyfin-web/blob/master/src/bundle.js#L176) in the web client. This then sends a copy of the login information to the python backend, which then [creates another session](https://github.com/iwalton3/jellyfin-mpv-shim/blob/master/jellyfin_mpv_shim/webclient_view/__init__.py#L72). There is also code in the web client that effectively automatically "casts" to the python backend.

If the python backend cannot connect but the web client does, then the web client tries to play the media in whatever player comes bundled with the webview I'm using. I've actually thought about completely breaking non-cast playback in the web client, just to make this error condition more visible.

**Comment by whalehub** [comment]
> This is a desktop debug build (so you don't have to export logs to see errors) with the client timeout set to 30 seconds. I hope this helps!
> https://iwalton.com/ushare/jellyfin-mpv-shim_1.7.0_timeout30.exe

Thanks for the quick build! Unfortunately it reports the same error message (with the same timeout value) in the CMD console:

```
2020-08-12 08:05:34,373 [   ERROR] JELLYFIN.jellyfin_apiclient_python.api: HTTPSConnectionPool(host='<redacted>', port=443): Read timed out. (read timeout=5)                                                                                                                                                  2020-08-12 08:05:34,373 [    INFO] JELLYFIN.jellyfin_apiclient_python.connection_manager: Failed to login as `<redacted>`
```

**Comment by iwalton3** [comment]
That would be because the connection manager has a separate copy of the API class... I'm building another version, hopefully this one works.

**Comment by iwalton3** [comment]
Ok. This version has a 30 second timeout for auth and a 5 second timeout for everything else.

https://iwalton.com/ushare/jellyfin-mpv-shim_1.7.0_atimeout30.exe

**Comment by whalehub** [comment]
> Ok. This version has a 30 second timeout for auth and a 5 second timeout for everything else.
> 
> https://iwalton.com/ushare/jellyfin-mpv-shim_1.7.0_atimeout30.exe

I can confirm that this fixes the issue. I did five test login attempts with this build and it worked every time (no error message, a `cred.json` file is generated and media playback works). The same goes for one of my users who tried it out. Thank you very much!

**Comment by iwalton3** [comment]
Ok I’ll make another test version tomorrow with a higher read timeout. (I’ll leave the connect timeout to 5 seconds, with the assumption it could be a slow password hash?)

I’d rather not use a 30 second connect timeout, as a user typing a bad URL could get stuck waiting up to 30 seconds to try again.

**Comment by iwalton3** [comment]
This version uses a 5 second connect timeout and a 30 second read timeout. It would provide a better user experience, so hopefully it works. Please let me know, if this one works I'll get a release out this evening.

https://iwalton.com/ushare/jellyfin-mpv-shim_1.7.0_loginc5r30.exe

**Comment by whalehub** [comment]
> This version uses a 5 second connect timeout and a 30 second read timeout. It would provide a better user experience, so hopefully it works. Please let me know, if this one works I'll get a release out this evening.
> 
> https://iwalton.com/ushare/jellyfin-mpv-shim_1.7.0_loginc5r30.exe

@iwalton3 Just did finished doing five login tests with the new build and asked a friend to test it as well and it consistently worked for both of us. 👍

**Comment by iwalton3** [comment]
This was released in 1.7.1.
